src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -362,25 +364,72 @@ public Info info() throws DockerException, InterruptedException {
     WebTarget resource = resource()
         .path("containers").path("json");
 
+    final Map<String, List<String>> filters = newHashMap();
     for (ListContainersParam param : params) {
-      resource = resource.queryParam(param.name(), param.value());
+      if (param instanceof ListContainersFilterParam) {
```

Comment:
Perhaps move this into a private method and use it in `public List<Image> listImages(final ListImagesParam... params)` as well?